llpc/tool/amdllpc.cpp

```diff
@@ -452,54 +452,72 @@ static Result Init(
             }
         }
 
-        // Initialize the path for shader cache
-        constexpr uint32_t MaxFilePathLen = 512;
-        char               shaderCacheFileDirOption[MaxFilePathLen];
+        const char* pName = "-shader-cache-file-dir";
+        size_t nameLen = strlen(pName);
+        bool found = false;
+        const char* pArg = nullptr;
+        for (int32_t i = 1; i < argc; ++i)
+        {
+            pArg = argv[i];
+            if ((strncmp(pArg, pName, nameLen) == 0) &&
+                ((pArg[nameLen] == '\0') || (pArg[nameLen] == '=')))
+            {
+                found = true;
+                break;
+            }
+        }
 
-        // Initialize the root path of cache files
-        // Steps:
-        //   1. Find AMD_SHADER_DISK_CACHE_PATH to keep backward compatibility.
-        const char* pEnvString = getenv("AMD_SHADER_DISK_CACHE_PATH");
+        if (found == false)
+        {
+            // Initialize the path for shader cache
+            constexpr uint32_t MaxFilePathLen = 512;
+            char               shaderCacheFileDirOption[MaxFilePathLen];
+
+            // Initialize the root path of cache files
+            // Steps:
+            //   1. Find AMD_SHADER_DISK_CACHE_PATH to keep backward compatibility.
+            const char* pEnvString = getenv("AMD_SHADER_DISK_CACHE_PATH");
```

Comment:
llvm has functions that deal with env: https://llvm.org/doxygen/classllvm_1_1sys_1_1Process.html. 
`llvm::GetEnv` returns an optional string with the value. Can you use that instead?
If this is not consistent with the rest of the tool, this can be a separate PR.